app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt

```diff
@@ -174,6 +180,29 @@ class ChildMonitorFragment : BaseDaggerFragment(), ServiceConnection {
                     pulsatingView.stop()
             }
         })
+
+        viewModel.pairingCodeLiveData.observe(viewLifecycleOwner, Observer { pairingCode ->
+            if (pairingCode.isNotEmpty()) {
+                showPairingDialog(pairingCode)
+            } else {
+                dismissPairingDialog()
+            }
+        })
+    }
+
+    private fun dismissPairingDialog() {
+        currentPairingDialog()?.dismiss()
+    }
+
+    private fun currentPairingDialog() =
+        (childFragmentManager.findFragmentByTag(PAIRING_CODE_DIALOG_TAG) as? DialogFragment)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/9259abfa8c2adcd3ca5c84779794ca9bdaf7e360/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment.kt#L198"></span>Parentheses in (childFragmentManager.findFragmentByTag(PAIRING_CODE_DIALOG_TAG) as? DialogFragment) are unnecessary and can be replaced with: childFragmentManager.findFragmentByTag(PAIRING_CODE_DIALOG_TAG) as? DialogFragment</td>
    </tr>
  </tbody>
</table>